int(math.Inf(1)) in funcOpMul is incorrect

This was caught out by a test failure on mips64:

```sh
--- FAIL: TestCliRun (0.86s)
    --- FAIL: TestCliRun/multiply_strings (0.00s)
        cli_test.go:87: standard output:
              (
              	"""
              	... // 8 identical lines
              	"abcabc"
              	null
            - 	null
            + 	""
              	"""
              )
            
FAIL
FAIL	github.com/itchyny/gojq/cli	0.872s
```

The result of `int(math.Inf(1))`<sup>[1][]</sup> depends on the architecture, and is not correct in general to do. 

On s390x / ppc64le (big-endian), it returns 9223372036854775807, which would have failed as well, would it not be for the limit condition.

A simple way to fix this is to check:

```go
math.IsNaN(cnt - cnt)
```

Which catches both infinite (positive / negative) as well as `NaN`.

[1]:https://github.com/itchyny/gojq/blob/main/operator.go#L381

--- a/operator.go
+++ b/operator.go
@@ -378,7 +378,7 @@ func funcOpMul(_, l, r interface{}) interface{} {
 		deepMergeObjects,
 		func(l, r interface{}) interface{} {
 			multiplyString := func(s string, cnt float64) interface{} {
-				if cnt <= 0.0 || int(cnt) < 0 || int(cnt) > maxHalfInt/(16*(len(s)+1)) {
+				if cnt <= 0.0 || int(cnt) > maxHalfInt/(16*(len(s)+1)) || math.IsNaN(cnt-cnt) {
 					return nil
 				}
 				if cnt < 1.0 {

